src/components/structures/MessagePanel.js

```diff
@@ -598,12 +605,21 @@ export default class MessagePanel extends React.Component {
         let isLastSuccessful = false;
         const isSentState = s => !s || s === 'sent';
         const isSent = isSentState(mxEv.getAssociatedStatus());
-        if (!nextEvent && isSent) {
+        const hasNextEvent = nextEvent && this._shouldShowEvent(nextEvent);
+        if (!hasNextEvent && isSent) {
             isLastSuccessful = true;
-        } else if (nextEvent && isSent && !isSentState(nextEvent.getAssociatedStatus())) {
+        } else if (hasNextEvent && isSent && !isSentState(nextEvent.getAssociatedStatus())) {
             isLastSuccessful = true;
         }
 
+        // This is a bit nuanced, but if our next event is hidden but a future event is not
+        // hidden then we're not the last successful.
+        if (nextEventWithTile) { // avoid length limit by wrapping in an if
+            if (isSentState(nextEventWithTile.getAssociatedStatus()) && nextEventWithTile !== nextEvent) {
```

Comment:
Since `nextEventWithTile !== nextEvent` will be faster to check, perhaps we can check it before the send state?

Reply:
The sent state should be pretty much the same number of nanoseconds